stdlib/string.gr

```diff
@@ -382,6 +382,32 @@ export let rec implode = (arr: Array<Char>) => {
   ret
 }
 
+/**
+ * Create a string that is the given string reversed.
+ *
+ * @param string: The string to reverse
+ * @returns A string whose characters are in the reverse order of the given string
+ *
+ * @example String.reverse("olleH") == "Hello"
+ *
+ * @since v0.5.0
+ */
+export let reverse = (string) => {
+  let len = wasmSafeLength(string)
+  let halfLen = len / 2
+  let mut arr = explode(string)
+  let mut i = 0
+  while (i < halfLen) {
```

Comment:
Should we use a `for` loop? I know a lot of the stdlib uses `while` because we wrote them before we had a for-loop, but I think we've been wanting to convert stuff over.

Reply:
Yeah, let's just use a for loop 🙂

Reply:
@technosophos we are hoping to get v0.4.5 released very soon. Any chance we could have this updated to land? Sorry to be a pain during the holiday week.